tools/generate-txn-import.js

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env node
+// @flow
+
+/* This script is run by Node directly, not via Babel; we can't use Flow
+   annotations inline (at least, not without jumping through additional hoops).
+
+   We can, however, use Flow's comment-annotation syntax [0]. So we do.
+
+   [0] https://flow.org/en/docs/types/comments/
+*/
+
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Given a sequence of strings, return a possibly-shorter list of unique strings in
+ * that list.
+ */
+const uniq /* : (Iterable<string>) => Array<string> */ = arr => {
+  const set = {};
+  const out = [];
+  for (const val of arr) {
+    if (!set[val]) {
+      out.push(val);
+    }
+    set[val] = true;
+  }
+  return out;
+};
+
+(async () => {
+  /** The project root. */
+  const root = path.resolve(__dirname, '..');
+  /** The human-readable name of this script. */
+  const thisScript = path.relative(root, __filename).replace(/^\.\//, '');
+
+  // ===================================================================
+  // Data acquisition
+  // ===================================================================
+
+  /** All the locales for which we have translation databases. */
+  const rawLocales /* : string[] */ = await (async () => {
+    // $FlowFixMe: Flow doesn't know about promises.opendir
```

Comment:
> // $FlowFixMe: Flow doesn't know about promises.opendir

Neither, apparently, does [Node 10](https://nodejs.org/docs/v10.0.0/api/fs.html#fs_fspromises_opendir_path_options), which I think is the latest version we explicitly require for anything (see 9f82dfb46bd66b30a97307783ba293f2deab1879).

So it might be nice to have a friendly error message if the version is less than [v12.12.0](https://nodejs.org/docs/v12.12.0/api/fs.html#fs_fspromises_opendir_path_options), or at least a comment saying that at least that version is required. I get this, currently:

```
$ node tools/generate-txn-import.js 
(node:83083) UnhandledPromiseRejectionWarning: TypeError: fs.promises.opendir is not a function
```

Reply:
Fixed – but I still get a warning:
```
(node:8216) ExperimentalWarning: The fs.promises API is experimental
```
... but I'm also running a somewhat older version of Node 10. If this is an issue for other people I can swap in the relevant `sync` calls instead.

Reply:
> I'm also running a somewhat older version of Node 10.

I'm puzzled that it would work on any version of Node 10; the docs at v12.12.0 say it was introduced in v12.12.0. 🤷‍♂